lib/scanner/index.js

```diff
@@ -41,43 +39,26 @@ class Scanner extends Emitter {
     this.onUpdate = options.onUpdate;
     this.getItems = options.getItems;
     this.onLost = options.onLost;
+
+    this.listener = DeviceEventEmitter
+      .addListener('magnetscanner:itemfound', this.onItemFound.bind(this));
   }
 
   start() {
     debug('start');
-    if (this.starting) return this.starting;
-    return this.starting = this.checkBluetooth()
-      .then(enabled => {
-        const stopped = !this.starting;
-        delete this.starting;
-
-        // protect against when .stop() is called between
-        // prompt dialog and actually starting the scanner
-        if (stopped) return Promise.reject('stopped');
-
-        // it's an error if the user denied the bluetooth prompt
-        if (!enabled) return Promise.reject('bt-disabled');
-
-        // actually start scanning now
-        MagnetScanner.start();
-        this.started = true;
-
+    return MagnetScanner.start()
+      .then(() => {
+        debug('started');
```

Comment:
Here we've effectively made the client dumber and the native side smarter. This is because the native side knows what its platform's dependencies are before the scanner can run, whereas the client is too abstract to make such decisions.

I found that we had non-trivial protection logic in both JS and native, it's best if we only have to maintain this stuff in one place. If the client attempts to call `.start()` when native side is 'busy', the promise will reject. But the client is aware that some errors are less serious than others.